test/performance/dataplane-probe/sla.go

```diff
@@ -26,75 +26,83 @@ import (
 	"knative.dev/pkg/test/mako"
 )
 
-var (
-	// This analyzer validates that the p95 latency talking to pods through a Kubernetes
-	// Service falls in the +5ms range.  This does not have Knative or Istio components
-	// on the dataplane, and so it is intended as a canary to flag environmental
-	// problems that might be causing contemporaneous Knative or Istio runs to fall out of SLA.
-	Kubernetes95PercentileLatency = &tpb.ThresholdAnalyzerInput{
+// This analyzer validates that the p95 latency talking to pods through a Kubernetes
+// Service falls in the +5ms range.  This does not have Knative or Istio components
+// on the dataplane, and so it is intended as a canary to flag environmental
+// problems that might be causing contemporaneous Knative or Istio runs to fall out of SLA.
+func newKubernetes95PercentileLatency(valueKey string) *tpb.ThresholdAnalyzerInput {
+	return &tpb.ThresholdAnalyzerInput{
 		Name: proto.String("Kubernetes baseline"),
 		Configs: []*tpb.ThresholdConfig{{
 			Min: bound(100 * time.Millisecond),
 			Max: bound(105 * time.Millisecond),
 			DataFilter: &mpb.DataFilter{
 				DataType:            mpb.DataFilter_METRIC_AGGREGATE_PERCENTILE.Enum(),
 				PercentileMilliRank: proto.Int32(95000),
-				ValueKey:            proto.String("kd"),
+				ValueKey:            proto.String(valueKey),
 			},
 		}},
 		CrossRunConfig: mako.NewCrossRunConfig(10),
 	}
+}
 
-	// This analyzer validates that the p95 latency talking to pods through Istio
-	// falls in the +8ms range.  This does not actually have Knative components
-	// on the dataplane, and so it is intended as a canary to flag environmental
-	// problems that might be causing contemporaneous Knative runs to fall out of SLA.
-	Istio95PercentileLatency = &tpb.ThresholdAnalyzerInput{
+// This analyzer validates that the p95 latency talking to pods through Istio
+// falls in the +8ms range.  This does not actually have Knative components
+// on the dataplane, and so it is intended as a canary to flag environmental
+// problems that might be causing contemporaneous Knative runs to fall out of SLA.
+func newIstio95PercentileLatency(valueKey string) *tpb.ThresholdAnalyzerInput {
```

Comment:
at this point why not pass the `Name` string and  bounds as well? 4 params should be tolerable.
then just 1 function to rule them all.

Reply:
I'm not sure if I correctly get your point, but is the new change what you want? I don't want to move the bounds and comments in the targets map since it looks clearer if defined separately.